docs/BR.md

```diff
@@ -2013,136 +2748,66 @@ If the signing key is P-521, the signature MUST use ECDSA with SHA-512. When enc
 
-a. __Certificate Field:__ `subject:commonName` (OID 2.5.4.3)  
-   __Required/Optional:__ Required  
-   __Contents:__ This field MUST be present and the contents SHOULD be an identifier for the certificate such that the certificate's Name is unique across all certificates issued by the issuing certificate.
+#### 7.1.4.2 Subject Attribute Encoding
 
-b. __Certificate Field:__ `subject:organizationName` (OID 2.5.4.10)  
-   __Required/Optional:__ Required  
-   __Contents:__ This field MUST be present and the contents MUST contain either the Subject CA's name or DBA as verified under [Section 3.2.2.2](#3222-dbatradename). The CA may include information in this field that differs slightly from the verified name, such as common variations or abbreviations, provided that the CA documents the difference and any abbreviations used are locally accepted abbreviations; e.g., if the official record shows "Company Name Incorporated", the CA MAY use "Company Name Inc." or "Company Name".
+This document defines requirements for the content and validation of a number of attributes that may appear within the `subject` field of a `tbsCertificate`. CAs SHALL NOT include these attributes unless their content has been validated as specified by, and only if permitted by, the relevant certificate profile specified within [Section 7.1.2](#712-certificate-content-and-extensions).
 
-c. __Certificate Field:__ `subject:countryName` (OID: 2.5.4.6)  
-   __Required/Optional:__ Required  
-   __Contents:__ This field MUST contain the two‐letter ISO 3166‐1 country code for the country in which the CA's place of business is located.
+Table: Attribute Encoding and Order Requirements
 
-### 7.1.5 Name constraints
+| __Attribute__            | __OID__    | __Specification__                               | __Encoding Requirements__ | __Max Length[^maxlength]__ |
+| ----                     | --         | ---                                             | ----                  | - |
+| `countryName`            | `2.5.4.6`  | [RFC 5280](https://tools.ietf.org/html/rfc5280) |                       | 2 |
+| `stateOrProvinceName`    | `2.5.4.8`  | [RFC 5280](https://tools.ietf.org/html/rfc5280) | MUST use `UTF8String` | 128 |
+| `localityName`           | `2.5.4.7`  | [RFC 5280](https://tools.ietf.org/html/rfc5280) | MUST use `UTF8String` | 128 |
+| `streetAddress`          | `2.5.4.9`  | X.520                                           | MUST use `UTF8String` | 128 |
+| `postalCode`             | `2.5.4.17` | X.520                                           | MUST use `UTF8String` | 40 |
+| `organizationName`       | `2.5.4.10` | [RFC 5280](https://tools.ietf.org/html/rfc5280) | MUST use `UTF8String` | 64 |
+| `surname`                | `2.5.4.4`  | X.520                                           | MUST use `UTF8String` | -  |
```

Comment:
Are you implying that there are no upper bounds for `surname` and `givenName` attributes? RFC 5280 restricts those attributes to 40 and 16 characters respectively but X.520 has lifted these restrictions. Is there an order of precedence (RFC 5280, X.520) for subjectDN upper-bound restrictions?

Reply:
I think you might be misreading RFC 5280. `ub-given-name-length` and `ub-surname-length` are only used in the context of the `PersonalName`, which is a member of the `BuiltInStandardAttributes` that are part of the X.400 `ORAddress`.

The length for these fields is defined by `X520Name`, by virtue of the following ASN.1 module:
```asn1
-- Naming attributes of type X520name

id-at-name                AttributeType ::= { id-at 41 }
id-at-surname             AttributeType ::= { id-at  4 }
id-at-givenName           AttributeType ::= { id-at 42 }
id-at-initials            AttributeType ::= { id-at 43 }
id-at-generationQualifier AttributeType ::= { id-at 44 }

-- Naming attributes of type X520Name:
--   X520name ::= DirectoryString (SIZE (1..ub-name))
--
-- Expanded to avoid parameterized type:
X520name ::= CHOICE {
      teletexString     TeletexString   (SIZE (1..ub-name)),
      printableString   PrintableString (SIZE (1..ub-name)),
      universalString   UniversalString (SIZE (1..ub-name)),
      utf8String        UTF8String      (SIZE (1..ub-name)),
      bmpString         BMPString       (SIZE (1..ub-name)) }
ub-name INTEGER ::= 32768
```

So yeah, I should update this to have an upper-bound, but the upper-bound is 32K.

Reply:
It seems that there are several people misreading 5280. I will create an issue in zlint pointing to this discussion which will hopefully update:
- https://github.com/zmap/zlint/blob/6d643b9bddd500d5585876859efa09403a1f0f42/v3/lints/rfc/lint_subject_given_name_max_length.go#L56
- https://github.com/zmap/zlint/blob/6d643b9bddd500d5585876859efa09403a1f0f42/v3/lints/rfc/lint_subject_surname_max_length.go#L56

Reply:
Filed https://github.com/zmap/zlint/issues/585

Reply:
While 5280 does specify the upper bound is 32K, X.520 (08/2005) indicates that the upper bound for givenName and surname is 64 characters. See the following email from PKIX WG: https://mailarchive.ietf.org/arch/msg/pkix/QHLrBfuhuUnjk6D5oWO2boyXL_4/, where there is some confusion on the value of ub-name in the RFC series.

Given this, there may be interoperability concerns if we specify the upper bound to 32K.

Reply:
Thanks for the reminder, Corey! So yeah, I think we're converging on the 2005 syntax of 64 characters, notwithstanding 5280?

Reply:
64 sounds good to me; that's what I had in my initial stab on the profiles work at https://github.com/digicert/validation-wg-cert-profiles/blob/main/tls.md.